File: api/Models/CartModel.cs
namespace api.Models
{
    public class CartItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public decimal TotalPrice { get; set; }
        public string ImageUrl { get; set; }
    }

    public class Cart
    {
        public List<CartItem> Items { get; set; }
        public decimal TotalAmount { get; set; }
    }

    public class Coupon
    {
        public string Code { get; set; }
        public decimal Amount { get; set; }
        public bool IsPercent { get; set; }
        public string Category { get; set; }
    }
    public class CheckoutModel
    {
        public Cart Cart { get; set; }
        public List<Coupon> Coupon { get; set; }
    }
}

File: client/src/app/models/cart.model.ts
import { Coupon } from "./coupon.model";

export interface CartItem {
    id: number;
    name: string;
    price: number;
    quantity: number;
    totalPrice: number;
    imageUrl: string;
    category: string;
}

export interface Cart {
    items: CartItem[];
    totalAmount: number;
}

export interface CheckoutModel {
    cart: Cart;
    isPoint: boolean;
    coupon?: Coupon[];
    pointAmount?: number;
}

// public class ResponseModel
// {
//     public decimal TotalPrice { get; set; }
//     public decimal Discount { get; set; }
//     public decimal FinalPrice { get; set; }
//     public decimal PointUsed { get; set; }
// }

export interface ResponseModel {
    totalPrice: number;
    discount: number;
    finalPrice: number;
    pointUsed: number;
}